docs/rfc/ethereum_attestation.md

```diff
@@ -0,0 +1,129 @@
+    }
+    ...
+}
+```
+
+## Ethereum smart contract
+
+A new Ethereum smart contract is deployed to the network, which lets users claim their Radicle IDs:
+
+```solidity
+contract Claims {
+    event Claimed(address indexed addr);
+    function claim(uint8 version, uint256 id) public;
+}
+```
+
+To claim an ID, call `claim` using your Ethereum account.
+It will emit an event `Claimed`, which later can be queried to discover your attestation.
+The claims have no expiration date and don't need to be renewed.
+
+Every new claim invalidates previous ones made with the same account.
+To only revoke a claim without creating a new one, use ID `0`,
+which is guaranteed to not match any existing identity.
+
+Currently supported `version` values:
+- `1` - an `id` is a SHA-1 root hash. The excess high bytes are zeros, e.g. for hash
+`fb3102b74d7254eed7f18a31a3ba1ea946bb1a99` the passed `id` is
+`000000000000000000000000fb3102b74d7254eed7f18a31a3ba1ea946bb1a99`
+- `2` - an `id` is a SHA-256 root hash
+
+We need to deploy an official instance of the `Claims` smart contract and
+it must be used by all the users.
+If anybody makes a claim using a different instance, it won't be recognized by others.
+
+## Creation of an attestation
+
+You need to perform 2 actions in any order:
+- Add or update an `https://radicle.xyz/ethereum/claim/v1` entry in your identity JSON.
+The entry's `address` must be your Ethereum address.
+It's highly recommended to set a short expiration date as Ethereum claims don't expire.
+- Call `claim` in the `Claims` smart contract. The `id` must point to your link identity.
+
+## Discovery from an Ethereum address
+
+When you have an Ethereum address, you can find the claimed link ID using an Ethereum client.
+The example calls are based on the standard [client JSON RPC API][rpc] and should be exposed
+by your favourite Ethereum client library.
+It's important that the client must be trusted not to hide the events.
+
+- Use [getLogs][rpc-logs] to get the newest `Claimed` event filtered for the given ethereum address
+- Get the event's `transactionHash` field and use it to fetch the transaction which emitted it with
+[getTransactionByHash][rpc-tx]
+- Validate that the transaction signature matches its data and the ethereum address.
+For reference the signature payload content is listed [here][rpc-sign].
+- Read the link ID from the transaction data
+- Verify that the link ID claims back the Ethereum address,
+see [Discovery from a Link ID](#discovery-from-a-link-id)
+
+## Discovery from a Link ID
+
+When you have a link ID, you can find the claimed Ethereum address.
+Obtain the tip of its identity chain and read the ethereum address from the identity JSON
+`address` field in section `https://radicle.xyz/ethereum/claim/v1`, unless it's expired.
+You need to verify that the given Ethereum address claims back the link ID,
+see [Discovery from an Ethereum address](#discovery-from-an-ethereum-address).
+
+## Revocation of an attestation
```

Comment:
It might be good to call out here that a revocation will only be as good as the propagation of the document in the network. What I mean is that if I revoke my address, I still have to wait for others to fetch those changes -- so _could_ still end up transferring MONIEZ to the old address.

Reply:
Added